spark/src/main/scala/org/apache/spark/sql/delta/DeltaOperations.scala

```diff
@@ -294,14 +294,15 @@ object DeltaOperations {
   case class CreateTable(
       metadata: Metadata,
       isManaged: Boolean,
-      asSelect: Boolean = false
+      asSelect: Boolean = false,
+      clusterBy: Option[Seq[String]] = None
   ) extends Operation("CREATE TABLE" + s"${if (asSelect) " AS SELECT" else ""}") {
     override val parameters: Map[String, Any] = Map(
       "isManaged" -> isManaged.toString,
       "description" -> Option(metadata.description),
       "partitionBy" -> JsonUtils.toJson(metadata.partitionColumns),
       "properties" -> JsonUtils.toJson(metadata.configuration)
-    )
+    ) ++ (clusterBy.map(CLUSTERING_PARAMETER_KEY -> JsonUtils.toJson(_)))
```

Comment:
Can this go inside the above `()`?

Reply:
Shouldn't this parameter only populated when clusterBy is specified?

Reply:
Hi @ami7o , when `clusterBy` is None, the right side of `++` will become `None` so it became a no-op.

Reply:
Hi @imback82 , the only way I could think of is to do:
```
    override val parameters: Map[String, Any] = (Seq(
      "isManaged" -> isManaged.toString,
      "description" -> Option(metadata.description),
      "partitionBy" -> JsonUtils.toJson(metadata.partitionColumns),
      "properties" -> JsonUtils.toJson(metadata.configuration)) ++
      (clusterBy.map(CLUSTERING_PARAMETER_KEY -> JsonUtils.toJson(_)))).toMap
```

But I don't particularly like it. Let me know if you prefer this.

Reply:
I think we should be consistent with `paritionBy` when there is no `PARTITIONED BY` clause?

Reply:
Good idea. I updated the PR to use this new behavior: always have `clusterBy` even if there's no CLUSTER BY clause.

Reply:
It seems that you forgot to remove `++ (clusterBy.map(CLUSTERING_PARAMETER_KEY -> JsonUtils.toJson(_)))` at line 306.